### PR TITLE
add slack notification feature to tekton tasks and pipelines

### PR DESCRIPTION
**Pull request by hakuna-matatah**
Issue #, if available:

Description of changes:
Add slack notification feature to tekton tasks and tekton pipelines 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/pipelines/eks/awscli-cl2-load.yaml [hunk 1]
```diff
@@ -14,7 +14,18 @@ spec:
   - name: nodes-per-namespace
   - name: cl2-load-test-throughput
   - name: results-bucket
+  - name: slack-hook
+  - name: slack-message
```
Comment (prateekgogia) [inline tests/pipelines/eks/awscli-cl2-load.yaml:18]: Do we need to pass message as a param to pipeline? can we construct this message in the pipeline and send as a param to underlying task? The reason is we are accepting one message string in the pipeline, how do we differentiate if this is for test getting started / deleted?

File: tests/tasks/notifications/slack.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  - name: message
+    default: 'default'
+    description: The message you want to post.
+  steps:
+  - name: send-notification
+    image: alpine/k8s:1.22.6
+    script: |
+      echo "Approving KCM requests"
+      kubectl certificate approve $(kubectl get csr | grep "Pending" | awk '{print $1}')  2>/dev/null || true
+      ENDPOINT_FLAG=""
```
Comment (prateekgogia) [inline tests/tasks/notifications/slack.yaml:22]: `ENDPOINT_FLAG` not used?

File: tests/pipelines/eks/awscli-cl2-load.yaml (hunk 1 above)
Comment (hakuna-matatah) [reply tests/pipelines/eks/awscli-cl2-load.yaml:18]: > Do we need to pass message as a param to pipeline? can we construct this message in the pipeline and send as a param to underlying task? The reason is we are accepting one message string in the pipeline, how do we differentiate if this is for test getting started / deleted?

Message here is going to be the job-name (1k/5k/10/15k/some other test name) that we want to post. We will prepend and append text to it in the tasks based on where we are using it like `<Job-name> got kicked off`. Not the complete message itself. 
Task/pipeline itself will be unaware of the meaningful job name itself that we want to see in slack notifications.

File: tests/tasks/notifications/slack.yaml (hunk 2 above)
Comment (hakuna-matatah) [reply tests/tasks/notifications/slack.yaml:22]: haha, it's not required here. Copy paste error :D. Good catch!

File: tests/pipelines/eks/awscli-cl2-load.yaml [hunk 3]
```diff
@@ -14,7 +14,18 @@ spec:
...
   tasks:
+  - name: slack-notification
+    params:
+    - name: slack-hook
+      value: $(params.slack-hook)
+    - name: slack-message
+      value: $(params.message)
```
Comment (prateekgogia) [inline tests/pipelines/eks/awscli-cl2-load.yaml:25]: It should be `$(params.slack-message)`

File: tests/pipelines/eks/awscli-cl2-load.yaml [hunk 4]
```diff
@@ -14,7 +14,18 @@ spec:
...
   - name: cl2-load-test-throughput
   - name: results-bucket
+  - name: slack-hook
+  - name: slack-message
   tasks:
+  - name: slack-notification
+    params:
+    - name: slack-hook
+      value: $(params.slack-hook)
+    - name: message
```
Comment (prateekgogia) [inline tests/pipelines/eks/awscli-cl2-load.yaml:24]: I would suggest keeping the naming consistent everywhere as `slack-message`

**Review by prateekgogia — APPROVED** [review]

**Pull request merged**